app/controllers/artefacts_controller.rb

```diff
@@ -57,6 +57,7 @@ def edit
   end
 
   def create
+    @artefact.content_id = SecureRandom.uuid
```

Comment:
Went for the explicit route: we always assign the `content_id` for `publisher` artefacts.